modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,577 @@
+    private static final String TEST_USER_GMAIL_COM = "test.user@gmail.com";
+    private static final String APPLICATION_AUDIENCE = "APPLICATION";
+    private static final String TEST_ROLE_APPLICATION = "test_role_application";
+    private static final String INTERNAL_ACTION_MANAGEMENT_VIEW = "internal_action_mgt_view";
+    private static final String INTERNAL_ACTION_MANAGEMENT_CREATE = "internal_action_mgt_create";
+    private static final String INTERNAL_ACTION_MANAGEMENT_UPDATE = "internal_action_mgt_update";
+    private static final String INTERNAL_ACTION_MANAGEMENT_DELETE = "internal_action_mgt_delete";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_LIST = "internal_org_user_mgt_list";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_VIEW = "internal_org_user_mgt_view";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_CREATE = "internal_org_user_mgt_create";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_UPDATE = "internal_org_user_mgt_update";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_DELETE = "internal_org_user_mgt_delete";
+    private static final String INTERNAL_APPLICATION_MANAGEMENT_VIEW = "internal_application_mgt_view";
+    private static final String INTERNAL_APPLICATION_MANAGEMENT_UPDATE = "internal_application_mgt_update";
+    private static final String INTERNAL_API_RESOURCE_VIEW = "internal_api_resource_view";
+    private static final String INTERNAL_API_RESOURCE_CREATE = "internal_api_resource_create";
+    private static final String CUSTOM_SCOPE_1 = "test_custom_scope_1";
+    private static final String CUSTOM_SCOPE_2 = "test_custom_scope_2";
+    private static final String CUSTOM_SCOPE_3 = "test_custom_scope_3";
+    private static final String SCIM2_USERS_API = "/o/scim2/Users";
+    private static final String ACTIONS_API = "/api/server/v1/actions";
+    private static final String APPLICATION_MANAGEMENT_API = "/api/server/v1/applications";
+    private static final String API_RESOURCE_MANAGEMENT_API = "/api/server/v1/api-resources";
+    private final static Boolean REQUIRES_AUTHORIZATION = true;
+    protected OAuth2ServiceHelper oAuth2Service;
+
+    protected SCIM2RestClient scim2RestClient;
+    private String roleID;
+    private String userID;
+    private String appId;
+
+    private String consumerKeyStr;
+    private String consumerSecretStr;
+    private String accessToken;
+
+    private String[] customScopesFromToken;
+
+    private List<String> consumerKeys = new ArrayList<>();
+    private List<String> consumerSecrets = new ArrayList<>();
+    private List<String> customScopes = new ArrayList<>();
+    private List<Permission> userPermissions = new ArrayList<>();
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.TENANT_USER);
+
+        oAuth2Service = this.new OAuth2ServiceHelper();
+
+        restClient = new ActionsRestClient(serverURL, tenantInfo);
```

Comment:
```suggestion
        actionsRestClient = new ActionsRestClient(serverURL, tenantInfo);
```

Reply:
Also restClient is initialized in the super class right ? Do we need to re initialize it ?

Reply:
Fixed with the commit